shared/fs/common/hooks.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react'
+import * as Container from '../../util/container'
+import * as Types from '../../constants/types/fs'
+import * as Constants from '../../constants/fs'
+import * as FsGen from '../../actions/fs-gen'
+import * as RPCTypes from '../../constants/types/rpc-gen'
+import uuidv1 from 'uuid/v1'
+
+const isPathItem = (path: Types.Path) => Types.getPathLevel(path) > 2 || Constants.hasSpecialFileElement(path)
+
+const useDispatchWhenConnected = () => {
+  const kbfsDaemonConnected =
+    Container.useSelector(state => state.fs.kbfsDaemonStatus.rpcStatus) ===
+    Types.KbfsDaemonRpcStatus.Connected
+  const dispatch = Container.useDispatch()
+  return kbfsDaemonConnected ? dispatch : () => {}
```

Comment:
you may want to make this empty call a const and hoist it so it doesn't change which will trash the hooks that use it as a dep

Reply:
did you put some logging in and verify this all works as intended and is only firing when you expect it to?

Reply:
I played around and tried a few things and they worked. Will inspect the log a bit to make sure we are not doing more than necessary

Reply:
Actually found a bug! I was loading all Tlfs instead of synced ones. Fixed in latest commit with a load-only component since we can't branch hooks.